addie/processing/idl/run_sum_scans.py

```diff
@@ -94,15 +103,50 @@ def create_output_file(self):
         f.close()
         print("[LOG] created file %s" % _full_output_file_name)
 
+        # redpar file
+        for _label in self._runs:
+            f = open(self._runs[_label]["sam_name"] + ".redpar", "w")
```

Comment:
Two things here I am just noticing now:
 1) The file should be opened using a `with` statement, current implementation is actually an anti-pattern (https://docs.quantifiedcode.com/python-anti-patterns/maintainability/not_using_with_to_open_files.html)

 2) I was hoping we could get the example redpar file here so maybe we could write an actual test to make sure this works since right now, I have no idea if it works or not. It probably would help to break this up into two different parts: one that creates the output but doesn't write it to file and then another part that opens the file and writes the content. This would help since the first part could be a method / function that you write a normal test for where you check the contents of the output data structure for the redpar file and the second could be where you check the written file against a redpar target test file we keep in the repository. The current way it is written, everything is embedded into a larger function that we cannot test to make sure it works (or that we don't break it in the future when we move the table around)

Reply:
> Two things here I am just noticing now:
> 
> 1. The file should be opened using a `with` statement, current implementation is actually an anti-pattern (https://docs.quantifiedcode.com/python-anti-patterns/maintainability/not_using_with_to_open_files.html)
> 2. I was hoping we could get the example redpar file here so maybe we could write an actual test to make sure this works since right now, I have no idea if it works or not. It probably would help to break this up into two different parts: one that creates the output but doesn't write it to file and then another part that opens the file and writes the content. This would help since the first part could be a method / function that you write a normal test for where you check the contents of the output data structure for the redpar file and the second could be where you check the written file against a redpar target test file we keep in the repository. The current way it is written, everything is embedded into a larger function that we cannot test to make sure it works (or that we don't break it in the future when we move the table around)

- Codes cleaned up to cope with the anti-pattern issue.

- Routine separated into two parts and implemented as functions for each part. I am not sure how to write a test for this. Could you please help with this? Thank you @marshallmcdonnell!

Reply:
I think you have already done plenty to make it testable, thanks!
We can just push it to another PR but the structure is now much better to write a test for it.

Reply:
No problem at all! So this PR should be ready to be merged, right? I have done the test locally on analysis and everything is working perfectly.